lib/internal/test_runner/test.js

```diff
@@ -362,8 +362,39 @@ class Test extends AsyncResource {
   }
 
   matchesTestNamePatterns() {
-    return ArrayPrototypeSome(testNamePatterns, (re) => RegExpPrototypeExec(re, this.name) !== null) ||
-      this.parent?.matchesTestNamePatterns();
+    const matchesByNameOrParent = ArrayPrototypeSome(testNamePatterns, (re) =>
+      RegExpPrototypeExec(re, this.name) !== null ||
+      this.parent?.matchesTestNamePatterns(),
+    );
+
+    if (matchesByNameOrParent) {
+      return true;
+    }
+
+    const testNameWithAncestors = this.getTestNameWithAncestors();
+    if (!testNameWithAncestors) {
+      return false;
+    }
+
+    return ArrayPrototypeSome(testNamePatterns, (re) => RegExpPrototypeExec(re, testNameWithAncestors) !== null);
```

Comment:
using [RegExp.test](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test) would probably offer better performance because it returns just boolean, but it's not allowed by linter 😞

Reply:
Using `RegExp#test` would defeat the purpose of primordials (which is that the test runner code should not be affected by the user code)
https://github.com/nodejs/node/blob/b0443690f89816ca3b9b8dee7e73f4527b3ff3e0/doc/contributing/primordials.md#L657-L664

Reply:
Ok 👍🏼 

I was surprised that it is present in primordials but cannot be used ;)